### DISCUSSION TITLE
2021 Release Plan

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#2180] [Announcements]
I thought I should communicate my current planning regarding releases over the next few months in 2021.

Version 5.5.0 was released at the end of July.

CSLA version 5.5.0 is intended to be the last enhancement release of CSLA 5, with all subsequent work going into CSLA 6.

I expect the release of CSLA 6 to be in November, along with the release of .NET 6 from Microsoft. There will be pre-releases prior to that point.

**Comment by Chicagoan2016** [comment]
So hopefully we could just target .NET 6 and not worry about Core, Standard etc.? Even for Win forms applications? : )

**Reply by rockfordlhotka** [reply]
That's the goal from a CSLA user perspective, absolutely!

The CSLA projects/packages will need to _at least_ keep targeting .NETfx 4.8, netstandard2.0 and .NET <x> for a very long time though.

.NETfx 4.8 is necessary because it'll be _many_ years before everyone moves off legacy .NET.

netstandard2.0 is necessary for a while, because that covers all the .NET Core world as it has been for a few years. Hopefully people move off those and onto .NET <x> sooner than later.

And of course, we'll need to support .NET <x> over time, probably for multiple values of <x> based on Microsoft's LTS and current versions.

**Comment by rockfordlhotka** [comment]
I wrote a [blog post](https://blog.lhotka.net/2021/07/26/CSLA-5-and-6) that has some insight into CSLA 6 as well.

**Comment by rockfordlhotka** [comment]
Time for an update on 2021 progress.

I just released version 5.5.2, which fixes some bugs in version 5.5.1. CSLA 5 is (in my mind) now in maintenance mode, and all my time for the past few months has really been focused on the next big release.

CSLA 6 is finally nearing its first prerelease. 

I have all sorts of reasons why this release has developed slower than I'd have liked: the pandemic, my company being acquired, spending time with family. Still, I'm disappointed in myself for my lack of focus.

Right now today, CSLA 6 has working code in

* Blazor wasm
* Blazor server
* WPF
* Console apps
* Local data portal
* Remote data portal (via HttpProxy)

I want to get Razor Pages and Windows Forms samples running before the prerelease, and with any luck that'll happen of the long weekend so there is a prerelease next week.

**At which point I will need as much help as possible with testing CSLA 6!**

The scope of .NET has expanded over the years, and I just can't keep up with all the UI frameworks to ensure all the various scenarios work in all of them. So much of my current focus is on Blazor, that I'm finding myself struggling to remember how WPF and Windows Forms work 😅

**Reply by rockfordlhotka** [reply]
Based on earlier posts, I should clarify that CSLA 6 currently targets:

* .NET Framework 4.6.2
* .NET Framework 4.7.2
* .NET Framework 4.8
* .NET Standard 2.0
* .NET 5
* .NET 6

I'll keep netfx 4.6.2 and 4.7.2 in the list as long as they require little or no effort. Eventually Microsoft will make changes to .NET that are challenging to backport, at which point I'll drop those versions.

Anyone using .NET today, should have migration strategies to get their (your) code to .NET Framework 4.8 or .NET 6 within the next couple years.
